boxoffice/forms/discount.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+
+from flask import g
+from baseframe import __
+import baseframe.forms as forms
+from baseframe.forms.sqlalchemy import QuerySelectMultipleField
+from coaster.utils import getbool
+from ..models import DISCOUNT_TYPE, DiscountPolicy
+
+__all__ = ['DiscountPolicyForm', 'DiscountCouponForm']
+
+
+class DiscountPolicyForm(forms.Form):
+    title = forms.StringField(__("Discount Title"),
+        validators=[forms.validators.DataRequired((u"Please specify a discount title")),
+        forms.validators.Length(max=250), forms.validators.StripWhitespace()])
+    is_price_based = forms.RadioField(__("Price based discount"),
+        choices=[(True, 'Special price discount'),
+        (False, 'Percentage based discount')], coerce=getbool, default=True)
+    discount_type = forms.RadioField(__("Discount Type"),
+        choices=DISCOUNT_TYPE.items(), coerce=int, default=1)
+    percentage = forms.IntegerField(__("Percentage"),
+        validators=[forms.validators.OptionalIf('is_price_based', (u"Please specify a discount percentage"))])
+    item_quantity_min = forms.IntegerField(__("Minimum number of tickets"), default=1)
+    discount_code_base = forms.StringField(__("Discount Title"),
+        validators=[forms.validators.OptionalIfNot('discount_type',
+            (u"Please specify a discount code base")),
+        forms.validators.Length(max=20), forms.validators.StripWhitespace()])
+    bulk_coupon_usage_limit = forms.IntegerField(__("Number of times a bulk coupon can be used"), default=1)
+    price_title = forms.StringField(__("Price Title"),
+        validators=[forms.validators.OptionalIfNot('is_price_based',
+            (u"Please specify a price title")), forms.validators.Length(max=250),
+        forms.validators.StripWhitespace()])
+    amount = forms.IntegerField(__("Amount"),
+        validators=[forms.validators.OptionalIfNot('is_price_based', (u"Please specify the discounted price"))])
+    start_at = forms.DateTimeField(__("Price start date"), format='%d %b %Y %H:%M:%S',
+        timezone=lambda: g.user.timezone,
+        validators=[forms.validators.OptionalIfNot('is_price_based', (u"Please specify start date for the price"))])
+    end_at = forms.DateTimeField(__("Price end date"), format='%d %b %Y %H:%M:%S', timezone=lambda: g.user.timezone,
+        validators=[forms.validators.OptionalIfNot('is_price_based', (u"Please specify end date for the price"))])
+    items = QuerySelectMultipleField(__("Items"), get_label='title', query_factory=lambda: [],
+        validators=[forms.validators.DataRequired((u"Please select a item to which discount is to be applied"))])
+
+    def validate_discount_code_base(self, field):
+        discount_policy = DiscountPolicy.query.filter_by(discount_code_base=field.data).first()
+        if discount_policy and discount_policy != DiscountPolicy.query.filter_by(title=self.title.data).first():
```

Comment:
Is it required? discount_code_base has a unique constraint in the discount policy table not just in the organisation.